tron/api/resource.py

```diff
@@ -36,23 +37,33 @@ def default(self, o):
         if isinstance(o, datetime.date):
             return o.isoformat()
 
+        if isinstance(o, collections.KeysView):
+            return list(o)
+
         return super(JSONEncoder, self).default(o)
 
 
 def respond(request, response_dict, code=http.OK, headers=None):
     """Helper to generate a json response"""
     request.setResponseCode(code)
-    request.setHeader(b'content-type', b'text/json')
+    request.setHeader(b'content-type', b'text/json; charset=utf-8')
     for key, val in six.iteritems((headers or {})):
         request.setHeader(str(key), str(val))
     return str(
         json.dumps(response_dict, cls=JSONEncoder) if response_dict else "",
-    )
+    ).encode('utf-8')
+
+
+def get_string_arg(request, name):
```

Comment:
Maybe the encoding/decoding logic should be added to requestargs.get_string? Seems like name.encode() could apply to all of the functions there, too.